pkg/controller.v1/tensorflow/job_test.go

```diff
@@ -549,11 +549,19 @@ var _ = Describe("TFJob controller", func() {
 				ctx := context.Background()
 				name := fmt.Sprintf(jobNameTemplate, idx)
 				tc.tfJob.SetName(name)
+				tc.tfJob.CreationTimestamp = metav1.Now()
 
 				By("creating a TFJob")
 				Expect(reconciler.Create(ctx, tc.tfJob)).Should(Succeed())
 
-				initializeReplicaStatuses(&tc.tfJob.Status, kubeflowv1.TFJobReplicaTypeWorker)
+				// We need to wait for synchronizing cache.
+				By("getting a created TFJob")
+				var updatedTFJob kubeflowv1.TFJob
+				Eventually(func() error {
+					return reconciler.Get(ctx, client.ObjectKeyFromObject(tc.tfJob), &updatedTFJob)
+				}, timeout, interval).Should(BeNil())
```

Comment:
We should get the latest TFJob after creating and updating TFJobs.

Reply:
Good one. Is this guaranteed to have cache synchronisation and we get the latest object?

Reply:
```suggestion
				By("getting a created TFJob")
				var updatedTFJob kubeflowv1.TFJob
				Expect(reconciler.Get(ctx, client.ObjectKeyFromObject(tc.tfJob), &updatedTFJob)).Should(Succeeded())
```

First, I tried the above code. But I couldn't obtain any TFJob. So we must wait for synchronizing cache.

Reply:
https://github.com/kubeflow/training-operator/commit/825cc27e2d198f3d2e1861ae3c4cbd33b77a2062

Reply:
I believe that we can avoid the following error since getting TFJob after creating one.

```
  Expected success, but got an error:
      <*errors.StatusError | 0xc0019a06e0>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "Operation cannot be fulfilled on tfjobs.kubeflow.org \"test-bof-0\": the object has been modified; please apply your changes to the latest version and try again",
              Reason: "Conflict",
              Details: {
                  Name: "test-bof-0",
                  Group: "kubeflow.org",
                  Kind: "tfjobs",
                  UID: "",
                  Causes: nil,
                  RetryAfterSeconds: 0,
              },
              Code: 409,
          },
      }
```